stock_dock/models/stock_dock.py

```diff
@@ -22,12 +23,11 @@ class StockDock(models.Model):
     company_id = fields.Many2one(
         comodel_name="res.company",
         string="Company",
-        related="warehouse_id.company_id",
-        readonly=True,
-        store=True,
-        index=True,
+        default=lambda self: self.env.company,
```

Comment:
A dock is part of a warehouse. So the dock belongs to the company of the warehouse. There is no reason to change that.